examples/MicroserviceExample/WebApi/MessageSender.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Context.Propagation;
+using RabbitMQ.Client;
+
+namespace WebApi
+{
+    public class MessageSender
+    {
+        private static readonly ITextFormat TextFormat = new TraceContextFormat();
+
+        private readonly ActivitySource activitySource = new ActivitySource(nameof(MessageSender));
+        private readonly ILogger<MessageSender> logger;
+
+        public MessageSender(ILogger<MessageSender> logger)
+        {
+            this.activitySource = new ActivitySource(nameof(MessageSender));
+            this.logger = logger;
+        }
+
+        internal string PublishMessage(IModel channel, string queueName)
+        {
+            // Start an activity with a name following the semantic convention of the OpenTelemetry messaging specification.
+            // https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md#span-name
+            string activityName = $"{queueName} send";
+            using (var activity = activitySource.StartActivity(activityName))
```

Comment:
@alanwest Would you please update to pass `ActivityKind.Producer` on the `StartActivity` call?

Reply:
Yes, thanks for reviewing this with an eye on the spec!